crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -151,13 +146,128 @@ pub struct AlgorithmUpdaterV1 {
     pub second_to_last_profit: i128,
     /// The latest known cost per byte for recording blocks on the DA chain
     pub latest_da_cost_per_byte: u128,
-
+    /// Activity of L2
+    pub l2_activity: L2ActivityTracker,
     /// The unrecorded blocks that are used to calculate the projected cost of recording blocks
     pub unrecorded_blocks: BTreeMap<Height, Bytes>,
 }
 
-/// A value that represents a value between 0 and 100. Higher values are clamped to 100
+/// Because the DA gas price can increase even when no-one is using the network, there is a potential
+/// for a negative feedback loop to occur where the gas price increases, further decreasing activity
+/// and increasing the gas price. The `L2ActivityTracker` is used to moderate changes to the DA
+/// gas price based on the activity of the L2 chain.
+///
+/// For each L2 block, the activity is calculated as a percentage of the block capacity used. If the
+/// activity is below a certain threshold, the activity is decreased. The activity exists on a scale
+/// between 0 and the sum of the increase, hold, and decrease buffers.
+///
+/// e.g. if the decrease range is 50, the hold range is 50, and the increase range is 50:
+///
+/// 0<-- decrease range -->50<-- hold range -->100<-- increase range -->150
+///
+/// The current activity determines the behavior of the DA gas price.
+///
+/// For healthy behavior, the activity should be in the `increase` range.
 #[derive(serde::Serialize, serde::Deserialize, Debug, Clone, PartialEq)]
+pub struct L2ActivityTracker {
+    /// The _size_ of the range of the activity buffer that will increase the gas price
```

Comment:
I would also suggest consistent usage of "DA gas price" and "L2 gas price". I was even thinking at some point about newtypes to avoid mistakes - both in implementation and in attempts to understand the algo :-)

Reply:
Not a bad idea with the types.

Reply:
Yes, definitely not in this PR, if at all.

Reply:
btw: would you mind leaving a `TODO` or creating a dedicated (good first) issue?

Reply:
Just don't leave any `TODOs` without referencing a ticket. I'm okay with a ticket without a TODO though.

Reply:
> Just don't leave any TODOs without referencing a ticket. I'm okay with a ticket without a TODO though. 
We should do this consistently :) 
It looks like there is a rustfmt setting that would be useful to enforce this in the CI: https://github.com/rust-lang/rustfmt/blob/23f01eddacccf4c9b871b0fd84d4344fd67bf668/src/config.rs#L386-L389. But I suspect it would be a massive job to go through the whole code-base and add issues for any todo.

Reply:
https://github.com/FuelLabs/fuel-core/issues/2382

Added the TODO with the issue attached.